### PR TITLE
Fix json decoding errors on grades and submissions

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14895
affects: student, teacher
release note: Fixed loading grades and submissions

test plan: see ticket

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



[MBL-14895](https://instructure.atlassian.net/browse/MBL-14895)

#### Affected Apps: Student, Teacher

#### Release Note: 
Fixed loading grades and submissions
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55371276/p/11cfdae56ecef155d42c698e7e70e774">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55371298/p/2ab2aff1636bb5e2e8eff89da4119dff">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55371260/p/e0866eaf6940bce973beb2a258d3b8d9">
    
  </a>
</details>

File: Core/Core/AudioVideo/APIMediaComment.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ public struct APIMediaComment: Codable, Equatable {
     let content_type: String
     let display_name: String?
     let media_id: String
-    let media_type: MediaCommentType
+    let media_type: String
```
Comment (vanwagonet) [inline Core/Core/AudioVideo/APIMediaComment.swift:25]: Empty string? or was there some other value that we'll eventually want to understand?

File: Core/Core/Submissions/APISubmission.swift [hunk 2]
```diff
@@ -36,10 +36,10 @@ public struct APISubmission: Codable, Equatable {
     let group_id: ID?
     let group_name: String?
     let id: ID
-    let late: Bool
+    let late: Bool?
     let late_policy_status: LatePolicyStatus?
     let media_comment: APIMediaComment?
-    let missing: Bool
+    let missing: Bool?
```
Comment (vanwagonet) [inline Core/Core/Submissions/APISubmission.swift:42]: Good call here.

File: Core/Core/AudioVideo/APIMediaComment.swift (hunk 1 above)
Comment (armstrongnate) [reply Core/Core/AudioVideo/APIMediaComment.swift]: I added a comment about it in code. It was actually `video/*`. We default unknown types to `.video` so I figure that's okay. I did go down the path of checking if the raw value contains strings like "audio" or "video" but not sure if worth it or not.

**Review by vanwagonet — APPROVED** [review]
QA+1

**Pull request merged**